src/components/project/handlers/set-department-id.handler.ts

```diff
@@ -72,11 +72,12 @@ export class SetDepartmentId implements IEventHandler<SubscribedEvent> {
     const res = await this.db
       .query()
       .raw(
+        // convert returned value from float to int and then to string
         `
         MATCH (:Project {id: $projectId})-[:primaryLocation {active: true}]
               -()-[:fundingAccount {active: true}]
               -()-[:accountNumber {active: true}]-(node:Property)
-        RETURN node.value as prefix
+        RETURN toString(toInteger(node.value)) as prefix
```

Comment:
What do you think about just changing the as result type to be correct
```tsx
.asResult<{ prefix: number }>()
```
And then just converting it to a string in TS
```tsx
return res.prefix.toString();
```

I would actually kinda like to replace "department id prefix" with "funding account number" for clarity also....

Reply:
Yeah either way. I guess it's a bit cleaner. I just feel cooler doing it neo4j. I can change the name too.

Reply:
So JS converts floats ending in zero `1.0` automatically into numbers of 1. That's fun.

Reply:
Right because `1 === 1.0`. This is why the TS type is `number` instead of `integer` or `float`. JS doesn't have distinct types for these.

Reply:
Right.